harlow/sampling/sampling_baseclass.py

```diff
@@ -112,63 +172,80 @@ def observer(self, X):
                 f"dimension 0 but are {X.shape[0]} and {y.shape[0]} respectively."
             )
 
-        return y
+        return res
 
     # Returns True if the loop should finish
-    def stopping_criterium(self, iteration: int, max_iter: int) -> bool:
+    def _stopping_criterium(
+        self, iteration: int, max_iter: int, last_score: float
+    ) -> bool:
+        if self.stopping_score is not None:
+            logger.info(f"Evaluation metric score on provided testset: {last_score}")
+            if last_score <= self.stopping_score:
+                logger.info(f"Algorithm converged in {iteration} iterations")
+
         return iteration > max_iter
 
     @abstractmethod
-    def best_new_points(self, n) -> np.ndarray:
+    def _best_new_points(self, n) -> np.ndarray:
         pass
 
     @staticmethod
-    def generate_run_name():
+    def _generate_run_name():
         return shortuuid.uuid()
 
-    def loop_initialization(self):
-        # initialize surrogate
+    def _fit_models(self):
```

Comment:
Perhaps I understand it incorrectly, but here only one surrogate is handled (index 0). 
I guess somewhere here we should take care of the possibility of multiple surrogates (1 for each desired input-output combination). see also my notes elsewhere.